management/guardian.go

```diff
@@ -84,6 +84,13 @@ func (m *MultiFactorManager) List(opts ...RequestOption) (mf []*MultiFactor, err
 	return
 }
 
+// Delete an enrollment
+//
+// See: https://auth0.com/docs/api/management/v2#!/Guardian/delete_enrollments_by_id
+func (m *MultiFactorManager) Delete(id string) error {
```

Comment:
I think it would be better to add EnrollmentManager here while the API is for an enrollment rather than a factor.

ref: https://github.com/go-auth0/auth0/pull/157/files#diff-e2b523a5aacf304cdcd66ffc8c82c95a9deac4f83af0eb75380fecf23a57a8bfR105